### ISSUE TITLE
`uv <command> --help` should prioritise options specific to the command

### ISSUE THREAD

**Issue by shaunhegarty** [astral-sh/uv#6856]
This is only a minor annoyance but I keep running into it. 

I typically use the terminal in VS Code, docked at the bottom, with code on top.

When I run `uv <command> --help` I expect to at a minimum see the command specific options first, since I assume this is supposed to be the quick reference. 
This is not an issue when using `uv help <command>` since it scrolls to the top, however this is the rich help and the purpose is different I think. 

Example, I want remind myself quickly what options `uv run` has:
![image](https://github.com/user-attachments/assets/67aec19d-ce6e-49e5-8ce5-44021abbdb12)

The things I want to see are way up at the top.

**Comment by zanieb** [comment]
I'm not sure what to do here. All of the details about the specific command are on top. It seems pretty weird to move the command-specific options to the bottom? I agree it's not great though. Unfortunately we have a lot of options!

**Comment by shaunhegarty** [comment]
Yeah I understand there's a lot of options. Possibly there's nothing to be done here that's worth doing. 

I was confused as to why all the options from `sync` were included. I thought maybe they just got appended in any case.
Reading the rich help in more detail, I found this to be surprising, at least in the specific case of `uv run`.
> When used in a project, the project environment will be created and updated before invoking the command.

So it's essentially doing a `uv sync` before and thus all of its options are required. Seems like `--no-project` is the only `uv run` specific command so it possibly wasn't the best example. 

How about `uv tree` instead. Are the index, installer, build and cache options relevant for commands like this?

**Comment by zanieb** [comment]
Unfortunately all of the resolver options are relevant for most project operations because we make sure everything is up to date. I wonder if we should hide more option sections by default and require `uv help` to see the full listing.

**Comment by aswild** [comment]
> I wonder if we should hide more option sections by default and require `uv help` to see the full listing.

I'd love to see something like this. Show only command-specific options in `uv foo -h` and list everything in `uv foo --help` or `uv help foo`.  I think clap's `Arg::hide_short_help` method should be able to help with that.

Even with a full-height terminal on my monitor, Index/Resolver/Installer/Build/Cache/Python/Global Options sections take up about 2 screens worth of scrollback, and most of the time they're not relevant to a quick glance through "what can this specific subcommand do?"
